### ISSUE TITLE
slow cross-dc gossip when joining first node

### ISSUE THREAD

**Issue by patriknw** [akka/akka#23803]
When starting a new multi-dc cluster some nodes must join across to another dc and it takes unnecessary long time for such nodes to be visible as Up in other DCs.
That is probably because `cross-data-center-gossip-probability = 0.2`.

I think we should increase this probability when there is only a few nodes in the DC, i.e. in the initial joining phase. When alone it should probably be 1.0.

**Comment by chbatey** [comment]
Could also be a cause of https://github.com/akka/akka/issues/23775 where one of the nodes waits for nodes in other DCs to be removed

**Comment by chbatey** [comment]
What kind of times are you seeing? 

I tried it with a 9 node cluster, 3 nodes in each DC, timings for the 8 nodes to join to the first node and be up.

0.2 - 1-2s 
0.5 - 1-2s

For a 4 node cluster, 2 nodes in each DC:

0.2- 2-4s
0.5 - 1-1.5

All run locally so no real network delay. I ran each of these a ~10 times.

So we do appear to get some benefit for clusters < 5. Do you think it is worth hard coding it to 0.5 for clusters < 5 or do you think its worth making it configurable?

**Comment by patriknw** [comment]
The expected speedup is not for moving to Up, but for nodes in other DCs to see it as Up.
E.g. 
DC1: 3 nodes
DC2: 1 node joins a node in DC1. Here it should immediately do cross-dc gossip to DC1, since it has no peers in DC2. Don't know if it's too much to treat that case as probability 1.0.

I was thinking of a hard coded adjustment algorithm based on number of nodes in own DC. A simple such could be to just use 0.5 when size (in own DC) is < 5, as you suggest.

**Comment by patriknw** [comment]
noticing another thing; `isGossipSpeedupNeeded` should only count members in own DC

**Comment by patriknw** [comment]
Isn't `ageSortedTopOldestMembersPerDc` wrong? It is only collecting up to `crossDcConnections` number of nodes, but that doesn't mean that those are the oldest.

**Comment by chbatey** [comment]
Sorry that was what I was measuring (for all nodes to see all the nodes as up).

I've pushed a multi-jvm test for your scenario here: https://github.com/chbatey/akka/commit/5ab9379188c301dba4a25d5b874ef58cf6c9c218

Doesn't make a huge difference (similar to the numbers above) based on the probability. I added extra gossip logging and can see that it usually takes one of the nodes in DC1 to select remote (then it only has one option). If we only have a single node in the new DC it normally selects to gossip with no one which is obviously wrong.

I'll add the sliding probability and fix those other things ^
